plutus-core/cost-model/budgeting-bench/Benchmarks/Arrays.hs

```diff
@@ -0,0 +1,74 @@
+{-# LANGUAGE BlockArguments   #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Benchmarks.Arrays (makeBenchmarks) where
+
+import Prelude
+
+import Common (createOneTermBuiltinBench, createTwoTermBuiltinBenchElementwise)
+import Control.Monad (replicateM)
+import Criterion.Main (Benchmark)
+import Data.Vector.Strict (Vector)
+import Data.Vector.Strict qualified as Vector
+import PlutusCore.Builtin (mkTyBuiltin)
+import PlutusCore.Core (Type)
+import PlutusCore.Default (DefaultFun (IndexArray, LengthOfArray, ListToArray), DefaultUni)
+import PlutusCore.Name.Unique (TyName)
+import System.Random.Stateful (StdGen, UniformRange (uniformRM), runStateGen_)
+
+--------------------------------------------------------------------------------
+-- Benchmarks ------------------------------------------------------------------
+
+makeBenchmarks :: StdGen -> [Benchmark]
+makeBenchmarks gen =
+  [ benchLengthOfArray gen
+  , benchListToArray gen
+  , benchIndexArray gen
+  ]
+
+benchLengthOfArray :: StdGen -> Benchmark
+benchLengthOfArray gen =
+  createOneTermBuiltinBench LengthOfArray [tyArrayOfInteger] listOfArrays
```

Comment:
I think we need to use the `create<N>TermBudgetingBenchWithWrappers` here to wrap the array arguments, like [this](https://github.com/IntersectMBO/plutus/blob/a48ebd739c1da07bffa467c54445945e5412b472/plutus-core/cost-model/budgeting-bench/Benchmarks/Lists.hs#L119), although that will only really matter for `listToArray` since the others are constant time.  We have to use the same size measures here as are used in the denotations.

Try adding the wrappers and re-running the benchmarks and see if the cost of `listToArray` changes significantly.  If it does, you'll have to rebuild the cost model with the new results.

Reply:
I'm thinking that we should probably make the costed-by-length measure the default one and maybe do the same for lists (although maybe then we'll need a wrapper that measures the entire memory footprint for some of the other things involving lists).  I'll make an issue to look at that separately.

Reply:
> Try adding the wrappers and re-running the benchmarks and see if the cost of `listToArray` changes significantly.

Done. 

Without the `ListCostedByLength` wrapper.
![image](https://github.com/user-attachments/assets/af4afa54-521f-4eab-a163-3cf98a2c2c2d)

Wrapped
![image](https://github.com/user-attachments/assets/e0706536-23ea-40cc-8741-a58b3c8493ff)

Reply:
This gives very similar results to the unwrapped version, but that's because the inputs are lists containing integers of size 1, so the "deep" size is pretty much the same as the length.  This should be more robust though, since the results shouldn't change if we were to use lists of `data` say, whereas the previous version would have made `listToArray` cheaper because you could handle apparently larger inputs in the same amount of time.